src/services/EdFi.DataManagementService.Api.Tests.E2E/Features/SchoolYearValidation.feature

```diff
@@ -0,0 +1,203 @@
+@SchoolYear
+Feature: School Year Reference Validation
+  Reference validation on School Years
+
+  """
+  @StudentEducationOrganizationAssociation Hook that will create the following data for the required tests
+  createStudent(studentUniqueId)
+  createSchool(schoolId)
+  createCourse(courseCode, schoolId)
+  createSession(schoolId, schoolYear, sessionName)
+  createCourseOffering(localCourseCode, courseCode, schoolId, schoolYear, sessionName)
+  createSection(localCourseCode, schoolId, schoolYear, sessionName, sectionIdentifier)
+  """
+
+  @Ignore @StudentEducationOrganizationAssociation
+  Scenario: Try creating a resource using a valid school year
+    When sending a POST request to "/ed-fi/studentSectionAssociation" with body
+    """
+        {
+        "beginDate": "2024-04-25",
+        "sectionReference": {
+            "localCourseCode": "string",
+            "schoolId": 0,
+            "schoolYear": 0,
+            "sectionIdentifier": "string",
+            "sessionName": "string",
+        },
+        "studentReference": {
+            "studentUniqueId": "string",
+        }
+        }
+    """
+    Then the response code is 201
+    And the response body is
+    """
+    """
+
+  @Ignore @StudentEducationOrganizationAssociation
+  Scenario: Try creating a resource using an invalid school year
+    When sending a POST request to "/ed-fi/studentSectionAssociation" with body
+    """
+        {
+        "beginDate": "2024-04-25",
+        "sectionReference": {
+            "localCourseCode": "string",
+            "schoolId": 0,
+            "schoolYear": 0,
+            "sectionIdentifier": "string",
+            "sessionName": "string",
+        },
+        "studentReference": {
+            "studentUniqueId": "string",
+        }
+        }
+    """
+    Then the response code is 400
```

Comment:
I was wrong! After looking closely at the ODS/API, the response code should be 409 when a reference does not exist.

Reply:
`409 | Conflict.  The request cannot be completed because it would result in an invalid state.  See the response body for details.`